Controllers/__init__.py

```diff
@@ -26,7 +27,7 @@
 class template_controller(ABC):
     _has_optimizer = False
     # Define the computation library in your controller class or in the controller's configuration:
-    _computation_library: "type[ComputationLibrary]" = None
```

Comment:
Technically, we always work with a static reference to the computation library. That means we don't create instances of it, but always call the methods on the class itself. Therefore, the type of this attribute is `type`. This is really only for documentation. The code will work no matter what we put here.

Btw, the subscript type notation `"type[ComputationLibrary]"` in quotes is necessary for python<=3.8. In later versions one can use `type[ComputationLibrary]`, but we maintain backward compatibility for now